.github/workflows/test-e2e-desktop.yaml

```diff
@@ -0,0 +1,109 @@
+name: Test E2E Desktop Client
+
+on:
+  pull_request:
+    types: [labeled]
+  workflow_dispatch:
+
+jobs:
+  test:
+    if: ${{ github.repository == 'hashicorp/boundary-ui' && github.event.label.name == 'desktop' }}
+    name: Run Desktop E2E Tests
+    runs-on: ubuntu-22.04
+    env:
+      ENOS_VAR_boundary_edition: "ent"
+      ENOS_VAR_e2e_debug_no_run: true
+      ENOS_VAR_boundary_license_path: "./license.hclic"
+      SETUP_CLI: true
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          fetch-depth: 0
+
+      - name: Set up Node
+        uses: actions/setup-node@39370e3970a6d050c480ffad4ff0ed4d3fdee5af # v4.1.0
+        with:
+          node-version: 20
```

Comment:
Not sure if there's a better way to keep this node version in sync with other places

Reply:
I think this will just be like our other actions and we'll have to keep them in sync ourselves